packages/next/next-server/server/render.tsx

```diff
@@ -442,8 +445,16 @@ export async function renderToHTML(
     })
 
     if (isSpr && !isFallback) {
+      const previewData = tryGetPreviewData(req, res, previewProps)
```

Comment:
This is re-done here instead of passing down from `next-server` so it's available in serverless mode right? Might be good to add a comment so someone doesn't try optimizing by passing it down instead

Or maybe we could optimize it by passing it down and only calling it in render if we're in `serverless` mode since we've already decrypted the data once in `server` mode

Reply:
Commented that it's already cached.